File: boardinghouse_app/lib/models/favourite_boarding_house.dart
class FavouriteBoardingHouse {
  int? id;
  int? userId;
  int? boardingHouseId;
  String? status;

  FavouriteBoardingHouse({
    this.id,
    this.userId,
    this.boardingHouseId,
    this.status,
  });

  factory FavouriteBoardingHouse.fromJson(Map<String, dynamic> json) {
    return FavouriteBoardingHouse(
      id: json['id'],
      userId: json['user_id'],
      boardingHouseId: json['boarding_house_id'],
      status: json['status'],
    );
  }
}

File: boardinghouse_backend/app/Models/FavouriteBoardingHouse.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class FavouriteBoardingHouse extends Model
{
    use HasFactory;
    protected $fillable = [
        'user_id',
        'boarding_house_id',
        'status'
    ];

    /**
     * Mối quan hệ thuộc về với model User.
     *
     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
     */
    public function user()
    {
        return $this->belongsTo(User::class);
    }

    /**
     * Mối quan hệ thuộc về với model BoardingHouse.
     *
     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
     */
    public function boardingHouse()
    {
        return $this->belongsTo(BoardingHouse::class);
    }
}
